pennylane/_qubit_device.py

```diff
@@ -873,20 +884,24 @@ def adjoint_jacobian(self, tape):
                 d_op_matrix = operation_derivative(op)
 
             op.inv()
-            phi = self._apply_operation(phi, op)
+            # I'd like to  use op.adjoint() here
+            # then we don't have to re-invert the operation at the end
+            ket = self._apply_operation(ket, op)
 
             if op.grad_method is not None:
                 if param_number in tape.trainable_params:
-                    mu = self._apply_unitary(phi, d_op_matrix, op.wires)
 
-                    jac_column = np.array(
-                        [2 * dot_product_real(lambda_, mu) for lambda_ in lambdas]
-                    )
-                    jac[:, trainable_param_number] = jac_column
+                    ket_temp = self._apply_unitary(ket, d_op_matrix, op.wires)
+
+                    for kk, bra_ in enumerate(bras):
+                        jac[kk, trainable_param_number] = 2 * dot_product_real(bra_, ket_temp)
```

Comment:
`bras` is a list of arrays, rather than an array with an extra dimension.  I thought about making it an array with an additional dimension.  Think I'll go ahead and proceed with trying to make that work.

Reply:
Ah, yes, sorry I overlooked that! The multiplication should be saving more than the conversion to array costs, I'd hope :)